source/MaterialXContrib/MaterialXMaya/AEtemplate/AEMaterialXSurfaceTemplate.mel

```diff
@@ -4,6 +4,7 @@ global proc MaterialXSurface_CommandButtonsNew(string $nodeName)
     string $reloadCommand = "reloadMaterialXNode \"" + $nodeName + "\"";
     button -label "Edit" -command $editCommand editButton;
     button -label "Reload" -command $reloadCommand reloadButton;
+    button -label "Reload Libraries" -command "reloadMaterialXLibraries";
```

Comment:
Added a button to the Attribute Editor as well.

Reply:
Aren't libraries global to the plugin and not specific to the node. If so it feels strange to put it here.

Also as far as I can see it can have undesired latent side-effects if all that occurs is the library document is reloaded without trying to rebuild shaders. i.e. if something was wrong it would not be found until either the creation of a new node, or an existing node was updated. 

Let me know if this is incorrect understanding however. Thanks.

Reply:
Sure, libraries are global. Putting this button in the AE just looks like an easy UI solution. At least the button is discoverable there. Would you rather have this accessible via a MaterialX menu (as part of Maya's main menu)? If this is controversial, we could refrain from exposing anything in GUI and only have the command for now, until a UX person makes the call. It's mostly for our own use anyway, isn't it?

Also true that the reloaded libraries take effect for documents loaded afterwards only. I thought that was the intent and that this behavior would be intuitive. I guess the alternative is to reload all documents in the scene when libraries are reloaded. But that, in my opinion, sounds more like undesired side-effects because it's more implicit. If users want this behavior, a TD can write a script that would call `reloadMaterialXLibraries` first and then enumerate all MaterialX nodes and the scene and call `reloadMaterialXNode` on them.

By the way, the added benefit of the explicit design is that the user can try two instances of the same material with different library settings side-by-side in the viewport, for debugging purposes.

Reply:
To put it another way: we now have two commands, `reloadMaterialXLibraries` and `reloadMaterialXNode` with a nice separation of responsibilities, giving users explicit control over when to load what. This is also why we rely on optionVars instead of complicating commands with additional arguments. Conflating `reloadMaterialXLibraries` with `reloadMaterialXNode` would take away some of that control.

Now, GUI may be a different issue. Maybe this functionality is too complex for GUI, letting people shoot themselves in the foot.

Reply:
Removed the button as discussed.